Tests/SonarScanner.MSBuild.PreProcessor.Test/Infrastructure/MockObjectFactory.cs

```diff
@@ -47,12 +47,19 @@ internal class MockObjectFactory : IPreprocessorObjectFactory
     public MockObjectFactory(TestLogger logger) : this() =>
         Logger = logger;
 
-    public MockObjectFactory(bool withDefaultRules = true, string organization = null)
+    public MockObjectFactory(bool withDefaultRules = true, string organization = null, Dictionary<string, string> serverProperties = null)
     {
         Server = new(organization);
 
         var data = Server.Data;
         data.ServerProperties.Add("server.key", "server value 1");
+        if (serverProperties is not null)
```

Comment:
I will leave this - we'll need it anyways with https://sonarsource.atlassian.net/browse/SCAN4NET-486